Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp

```diff
@@ -15,7 +19,22 @@ RhoFunctor::operator() ( amrex::MultiFab& mf_dst, const int dcomp, const int /*i
     auto& mypc  = warpx.GetPartContainer();
 
     // Deposit charge density
-    std::unique_ptr<amrex::MultiFab> rho = mypc.GetChargeDensity( m_lev );
+    // Call this with local=true since the parallel transfers will be handled
+    // by ApplyFilterandSumBoundaryRho
+    std::unique_ptr<amrex::MultiFab> rho = mypc.GetChargeDensity( m_lev, true );
+
+    // Call this in case filtering is turned on
+    warpx.ApplyFilterandSumBoundaryRho(m_lev, m_lev, *rho, 0, rho->nComp());
+
+#ifdef WARPX_DIM_RZ
```

Comment:
Should this be
```suggestion
#ifdef WARPX_USE_PSATD
```
It seems that the code in the `ifdef` should be applicable for Cartesian PSATD, while on the other hand it should not work for RZ FDTD

Reply:
Thanks for checking this. This should be checking for both `WARPX_DIM_RZ` and `WARPX_USE_PSATD`. The `ApplyFilter` method is only defined in the RZ version of the spectral solver. I'll make the fix.